include/rmm/detail/dynamic_load_runtime.hpp

```diff
@@ -136,9 +136,16 @@ struct async_alloc {
   {
     int supported_handle_types_bitmask{};
 #if CUDART_VERSION >= 11030  // 11.3 introduced cudaDevAttrMemoryPoolSupportedHandleTypes
-    RMM_CUDA_TRY(cudaDeviceGetAttribute(&supported_handle_types_bitmask,
-                                        cudaDevAttrMemoryPoolSupportedHandleTypes,
-                                        rmm::detail::current_device().value()));
+    auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
+                                               cudaDevAttrMemoryPoolSupportedHandleTypes,
+                                               rmm::detail::current_device().value());
+
+    // Don't throw on cudaErrorInvalidValue
+    auto const unsupported_runtime = (result == cudaErrorInvalidValue);
+    if (unsupported_runtime) return false;
+    // throw any other error that may have occurred
+    RMM_CUDA_TRY(result);
```

Comment:
We should change this so it always returns true for `cudaMemHandleTypeNone` even if the driver is older than CUDA 11.3. (when `cudaDevAttrMemoryPoolSupportedHandleTypes` was introduced).

```suggestion
    if (cudaMemHandleTypeNone != handle_type) {
        auto const result = cudaDeviceGetAttribute(&supported_handle_types_bitmask,
                                                   cudaDevAttrMemoryPoolSupportedHandleTypes,
                                                   rmm::detail::current_device().value());

        // Don't throw on cudaErrorInvalidValue
        auto const unsupported_runtime = (result == cudaErrorInvalidValue);
        if (unsupported_runtime) return false;
        // throw any other error that may have occurred
        RMM_CUDA_TRY(result);
    }
```